Common/NearestNeighbour/src/main/java/org/tribuo/common/nearest/KNNModel.java

```diff
@@ -319,20 +284,7 @@
      * @return The predictions.
      */
     private List<Prediction<T>> innerPredictWithinExampleThreadPool(Iterable<Example<T>> examples) {
-        BiFunction<SparseVector,SparseVector,Double> distanceFunc;
-        switch (distance) {
-            case L1:
-                distanceFunc = (a,b) -> b.l1Distance(a);
-                break;
-            case L2:
-                distanceFunc = (a,b) -> b.l2Distance(a);
-                break;
-            case COSINE:
-                distanceFunc = (a,b) -> b.cosineDistance(a);
-                break;
-            default:
-                throw new IllegalStateException("Unknown distance function " + distance);
-        }
+        BiFunction<SGDVector,SGDVector,Double> distanceFunc = (a,b) -> DistanceType.getDistance(b, a, distType);
```

Comment:
Maybe it's better to pass the enum through rather than use a lambda? Alternatively we could push the lambda into the enum itself, and remove the switch in `DistanceType`, so `getDistance` just calls the lambda stored as a field in the enum.